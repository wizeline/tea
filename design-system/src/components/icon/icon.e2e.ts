import { newE2EPage } from '@stencil/core/testing';

describe('wz-icon', () => {
  it('renders icon', async () => {
    const page = await newE2EPage();
    await page.setContent('<wz-icon icon="clear" />');
    const icon = await page.find('wz-icon');
    expect(icon.outerHTML).toMatchSnapshot();
  });
});
