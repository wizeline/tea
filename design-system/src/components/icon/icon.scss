wz-icon {
   --icon-fill: var(--info-light);
  .xsmall, .xsmall svg {
    width: 16px;
    height: 16px;
  }

  .small, .small svg {
    width: 24px;
    height: 24px;
  }

  .medium, .medium svg {
    width: 32px;
    height: 32px;
  }

  .large, .large svg {
    width: 48px;
    height: 48px;
  }

  .xlarge, .xlarge svg {
    width: 56px;
    height: 56px;
  }

  svg {
    fill: var(--icon-fill);
  }
}
