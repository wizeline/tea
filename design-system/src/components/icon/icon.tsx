import { Component, Prop, Element, h } from '@stencil/core';
import sizeMatcher, { MEDIUM_SIZE } from '../../matchers/size';
import IconSet from '../icons';

@Component({
  tag: 'wz-icon',
  styleUrl: 'icon.scss',
  shadow: false,
})
export class Icon {
  // Size props
  @Prop()
  xxsmall: boolean;
  @Prop()
  xsmall: boolean;
  @Prop()
  small: boolean;
  @Prop()
  medium: boolean;
  @Prop()
  large: boolean;
  @Prop()
  xlarge: boolean;
  @Prop()
  color: string;
  @Prop()
  icon: string;

  @Element()
  el: HTMLElement;
  render() {
    const SelectedIcon = IconSet(this);
    const size = sizeMatcher(this) || MEDIUM_SIZE;

    this.el.style.setProperty('--icon-fill', this.color);

    const className = ['icon-container', size].join(' ');

    return <div class={className}>{SelectedIcon}</div>;
  }
}
