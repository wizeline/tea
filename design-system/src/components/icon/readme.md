# wz-icon



<!-- Auto Generated Below -->


## Properties

#### color

string


#### icon

string


#### large

boolean


#### medium

boolean


#### small

boolean


#### xlarge

boolean


#### xsmall

boolean


#### xxsmall

boolean


## Attributes

#### color

string


#### icon

string


#### large

boolean


#### medium

boolean


#### small

boolean


#### xlarge

boolean


#### xsmall

boolean


#### xxsmall

boolean



----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
