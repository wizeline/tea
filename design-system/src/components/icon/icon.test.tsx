import { TestWindow } from '@stencil/core/testing';
import { Icon } from './icon';

describe('wz-icon', () => {
  it('renders icon', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [Icon],
      html: '<wz-icon icon="clear" />',
    });

    expect(element).toMatchSnapshot();
  });
});
