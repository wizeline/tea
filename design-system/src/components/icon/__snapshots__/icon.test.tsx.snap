// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-icon renders icon 1`] = `
<wz-icon
  class="hydrated"
  icon="clear"
>
  
  <div
    class="icon-container medium"
  >
    <icon-clear />
  </div>
</wz-icon>
`;
